File: Backend/ClothingEcommerceWebsite/src/main/java/com/project/ClothingEcommerceWebsite/models/Review.java
package com.project.ClothingEcommerceWebsite.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "reviews",
        uniqueConstraints = @UniqueConstraint(columnNames = {"product_id","user_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id", nullable = false)
    @JsonIgnore
    private Product product;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private Integer rating;

    @Column(length = 255)
    private String title;

    @Lob
    private String content;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: Frontend/client-website/src/types/review.ts
import { Order } from "./order";
export interface CreateReviewData {
  user_id: number | undefined;
  product_id: number;
  order_id?: number;
  rating: number;
  title?: string;
  content?: string;
}

export interface Review {
  id: number;
  product_id: number;
  user_id: number;
  order?: Order;
  order_id?: number;
  rating: number;
  title?: string;
  content?: string;
  is_approved: boolean;
  createdAt: string;
  user?: {
    id: number;
    fullName: string;
    email?: string;
  };
  product?: {
    id: number;
    name: string;
    slug: string;
  };
}
